modules/openapi-generator/src/main/resources/typescript-inversify/api.service.mustache

```diff
@@ -179,13 +181,13 @@ export class {{classname}} {
 {{/formParams}}
 
 {{/hasFormParams}}
-        const response: Observable<HttpResponse<{{#returnType}}{{{returnType}}}{{#isResponseTypeFile}}|undefined{{/isResponseTypeFile}}{{/returnType}}{{^returnType}}any{{/returnType}}>> = this.httpClient.{{httpMethod}}(`${this.basePath}{{{path}}}{{#hasQueryParams}}?${queryParameters.join('&')}{{/hasQueryParams}}`{{#bodyParam}}, {{paramName}} {{/bodyParam}}{{#hasFormParams}}, formData{{/hasFormParams}}, headers);
+        const response: {{#usePromise}}Promise{{/usePromise}}{{^usePromise}}Observable{{/usePromise}}<HttpResponse<{{#returnType}}{{{returnType}}}{{#isResponseTypeFile}}|undefined{{/isResponseTypeFile}}{{/returnType}}{{^returnType}}any{{/returnType}}>> = this.httpClient.{{httpMethod}}(`${this.basePath}{{{path}}}{{#hasQueryParams}}?${queryParameters.join('&')}{{/hasQueryParams}}`{{#bodyParam}}, {{paramName}} {{/bodyParam}}{{#hasFormParams}}, formData{{/hasFormParams}}, headers);
         if (observe === 'body') {
-               return response.pipe(
-                   map((httpResponse: HttpResponse) => <{{#returnType}}{{{returnType}}}{{#isResponseTypeFile}}|undefined{{/isResponseTypeFile}}{{/returnType}}{{^returnType}}any{{/returnType}}>(httpResponse.response))
-               ){{#usePromise}}.toPromise(){{/usePromise}};
+               return response.{{#usePromise}}then{{/usePromise}}{{^usePromise}}pipe{{/usePromise}}(
+                   {{^usePromise}}map({{/usePromise}}(httpResponse: HttpResponse) => <{{#returnType}}{{{returnType}}}{{#isResponseTypeFile}}|undefined{{/isResponseTypeFile}}{{/returnType}}{{^returnType}}any{{/returnType}}>(httpResponse.response){{^usePromise}}){{/usePromise}}
+               ){{^usePromise}}.toPromise(){{/usePromise}};
         }
-        return response{{#usePromise}}.toPromise(){{/usePromise}};
+        return response{{^usePromise}}.toPromise(){{/usePromise}};
```

Comment:
This doesnt look right

Reply:
It's not right :-) 
But the latest commit [d44893d](https://github.com/OpenAPITools/openapi-generator/pull/11249/commits/d44893dccab8f4da8368a85a42ac9160769117d4) did actually fix that.
Hate to sound like a noob, but I've never actually **re**-submitted a PR before.  Did I make a mistake in how I pushed that latest commit?

Reply:
It‘s fine. By adding more commits to a branch, the PR is updated accordingly and the accumulated changes can be reviewed/merged.